qt/python/mantidqt/widgets/plotconfigdialog/test/test_apply_all_properties.py

```diff
@@ -129,20 +145,22 @@ def _run_apply_properties_on_figure_with_image():
     return img_ax
 
 
-def _run_apply_properties_on_figure_with_legend():
+def _run_apply_properties_on_figure_with_legend(curve_view_mock):
     fig = figure()
     ax = fig.add_subplot(111)
     ax.plot([1, 2, 3], label='old label')
     legend = ax.legend()
     legend.get_frame().set_alpha(0.5)
+    curve_view_mock.get_selected_curve_name = CurveNameSideEffect('old label', 'New label', switch_count=3)
+
     with patch.object(AxesTabWidgetPresenter, 'update_view', mock_axes_tab_presenter_update_view):
         presenter = PlotConfigDialogPresenter(fig, view=Mock())
     with patch.object(presenter.tab_widget_presenters[1], 'update_view',
                       lambda: None):
         with patch.object(presenter.tab_widget_presenters[1], 'axis_changed',
                           lambda: None):
             presenter.apply_properties()
-    return ax
+    return ax, curve_view_mock
```

Comment:
Everything looks great but i dont understand why you are returning curve_view_mock if it is not used in the test.

Reply:
Good spot, I think this was left over from tinkering. removed in the next commit !